approov-service/src/main/java/io/approov/service/okhttp/DefaultMessageSigningConfigFactory.java

```diff
@@ -0,0 +1,195 @@
+package io.approov.service.okhttp;
+
+import java.util.ArrayList;
+import okhttp3.Request;
+
+/* Add the following interfaces */
```

Comment:
This file gives a rough implementation of what it should look like - it is not proper Java as I don't have a working build env. The interfaces and classes will need their own files because they are all public.

Also, all comments need to be fleshed out and some code needs to be implemented from the comments that are provided.

Overall though, this looks like a good solution which allows debugging on the server side if a flag is set in the client and otherwise minimises the server side configuration that is required to get the signture check working. It also enables us to extend the communicated data using the same header in the near future.